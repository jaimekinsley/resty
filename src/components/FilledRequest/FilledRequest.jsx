import React from 'react';
import PropTypes from 'prop-types';

const FilledRequest = ({ method, api }) => (
  <>
    <p>{method}</p>
    <p>{api}</p>
  </>
);

FilledRequest.propTypes = {
  method: PropTypes.string.isRequired,
  api: PropTypes.string.isRequired
};

export default FilledRequest;
