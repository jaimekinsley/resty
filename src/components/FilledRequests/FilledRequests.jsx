import React from 'react';
import PropTypes from 'prop-types';
import FilledRequest from '../FilledRequest/FilledRequest';

const FilledRequests = ({ filledRequests }) => {
  const filledRequestElements = filledRequests.map(filledRequest => (
    <li key={Date.now()}>
      <FilledRequest {...filledRequest} />
    </li>
  ));
  return (
    <ul>
      {filledRequestElements}
    </ul>
  );
};

FilledRequests.propTypes = {
  filledRequests: PropTypes.arrayOf(PropTypes.shape({
    method: PropTypes.string.isRequired,
    api: PropTypes.string.isRequired
  }))
};

export default FilledRequests;
