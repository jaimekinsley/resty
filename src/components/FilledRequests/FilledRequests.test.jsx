import React from 'react';
import { shallow } from 'enzyme';
import FilledRequests from './FilledRequests';
import FilledRequest from '../FilledRequest/FilledRequest';


describe('FilledRequests component', () => {
  let wrapper;
  beforeEach(() => {
    const filledRequests = [
      { method: 'get', api: 'http://firstrequest.com' },
      { method: 'put', api: 'http://secondrequest.com' }
    ];
    wrapper = shallow(<FilledRequests filledRequests={filledRequests} />);
  });

  it('displays a list of two FilledRequest components', () => {
    expect(wrapper.find(FilledRequest)).toHaveLength(2);
  });
});
