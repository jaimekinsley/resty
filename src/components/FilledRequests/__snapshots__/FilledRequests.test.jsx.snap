// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilledRequest component renders a FilledRequest 1`] = `
<Fragment>
  <p>
    get
  </p>
  <p>
    https://futuramaapi.com
  </p>
</Fragment>
`;

exports[`FilledRequests component renders a FilledRequest 1`] = `
<Fragment>
  <p>
    get
  </p>
  <p>
    https://futuramaapi.com
  </p>
</Fragment>
`;
